src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
         /// <param name="matrix">The source matrix to invert.</param>
         /// <param name="result">If successful, contains the inverted matrix.</param>
         /// <returns>True if the source matrix could be inverted; False otherwise.</returns>
-        public static bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
+        public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
         {
+            if (Sse.IsSupported)
```

Comment:
Given how big the individual implementations are, it might be good to move them to different methods:
```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static unsafe bool Invert(Matrix4x4 matrix, out Matrix4x4 result)
{
    if (Sse.IsSupported)
    {
        return SseImpl();
    }

    return SoftwareFallback();

    static bool SseImpl(Matrix4x4 matrix, out Matrix4x4 result)
    {
    }

    static bool SoftwareFallback(Matrix4x4 matrix, out Matrix4x4 reuslt)
    {
    }
}
```

We've done this for various other methods where the different paths are this big.

Reply:
I've broken up the two implementations.